app/uk/gov/hmrc/emcstfe/repositories/CreateMovementUserAnswersRepository.scala

```diff
@@ -124,11 +128,21 @@ class CreateMovementUserAnswersRepositoryImpl @Inject()(mongoComponent: MongoCom
       _ =>
         collection
           .updateOne(
-            filter = by(ern, draftId),
+            filter = byDraftId(ern, draftId),
             update = Updates.set("hasBeenSubmitted", false))
           .headOption()
           .map(_.exists(_.getModifiedCount == 1L))
     }
+
+  def setErrorMessagesForDraftMovement(ern: String, lrn: String, errors: Seq[MovementSubmissionFailure]): Future[Option[String]] =
```

Comment:
Q: how will this work if you have multiple drafts with the same LRN?

Reply:
Uses a `submittedDraftId` field which is retrieved from the submission failure message correlation ID